dlk/python/dlk/templates/src/func/mean.cpp

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2019 The Blueoil Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "global.h"
+#include "func/mean.h"
+#include "time_measurement.h"
+
+void func_Mean(const TensorView<T_FLOAT, MemoryLayout::NHWC>& input,
+    const TensorView<int, MemoryLayout::C>& factor,
+    const TensorView<T_FLOAT, MemoryLayout::NC>& output, 
+    T_UINT in_h, T_UINT in_w) {
+#ifndef RUN_AS_HLS
+  Measurement::Start("Mean");
+#endif
+  T_UINT in_size = in_h * in_w;
+  T_UINT out_depth = output.size();
+
+  for (T_UINT dh = 0; dh < in_h; dh++){
```

Comment:
There's no concept of `axis` here, which looks not consistent with python version.

Reply:
@tkng If we take `axis` into consideration, there will be various layout needed for the output. Do you have any better ideas to deal with this `axis` stuff.

Reply:
Do you mean that current output tye is only `TensorView<T_FLOAT, MemoryLayout::NC>& output`, we need to add some output type if we want to support `axis` argument?

I think we can use `MemoryLayout::NHWC` for both of input and output shape, (one dimension of `HWC` will be collapsed to just 1, as a result of computation.)

Also, it would be preferable to consider the behavior of when `axis` is `None`. In that case, TF reduce all dimensions. Should we support such a behavior explicitly? Other possibility is just rejecting such a case.

Reply:
How the new concat operator is implemented in #472 looks useful information to implement generarized reduced_mean operator.